oracle/price_spike_model.py

```diff
@@ -0,0 +1,128 @@
+def apply_ema(old_value, new_value, smoothing_factor):
```

Comment:
Can we use inheritance to define a parent oracle model class that these implementations can extend? (limits duplicate code)